documentation/development-docs/sdkbased-vs-autogen.md

```diff
@@ -0,0 +1,102 @@
+# Description
+
+The document is for those, who are familiar with SDK-based modules and want to give a try with auto-gen modules. The content will contain the comparation between these two kinds of modules from architecture, cmdlets differences, and onboard process, etc.
+
+# Architecture
+
+## SDK-based Module
+![image](images/sdkbased.PNG)
+
+## Auto-gen Module
+![image](images/autogen.PNG)
+
```

Comment:
I think we can put a table here

||SDK based| Auto-gen based|
|-|-|-|
|Programming language|C#|C# (generated), PowerShell (customization)|
|docs|Generated by PlatyPS| Generated by AutoRest, together with code|
|...|